dynamodb/src/main/scala/uk/gov/nationalarchives/DADynamoDBClient.scala

```diff
@@ -31,6 +31,30 @@ class DADynamoDBClient[F[_]: Async](dynamoDBClient: DynamoDbAsyncClient):
   extension [T](completableFuture: CompletableFuture[T])
     private def liftF: F[T] = Async[F].fromCompletableFuture(Async[F].pure(completableFuture))
 
+  /** Deletes a list of items with primary keys of type T from Dynamo. Unprocessed items will be retried
+    *
+    * @param tableName
+    *   The name of the table
+    * @param primaryKeys
+    *   A list of primary keys to delete from Dynamo (list can be any length; requests will be batched into groups of
```

Comment:
1. Since you're reusing the same code as the BatchWriteItem, does that mean you can delete keys like this

```
List(PrimaryKeys("key1": "value1", "key2": "value2"))
```
and also

```
List(
  PrimaryKey("key1": "value1"),
  PrimaryKey( "key2": "value2"),
)
```

as well as

```
List(
  PrimaryKeys("key1": "value1", "key2": "value2"),
  PrimaryKeys("key3": "value3", "key4": "value4")
)

```
because this is what the tests indicate
2. if so, should we specify a preference?

Reply:
I'm not sure what you mean with this. You pass in a list of primary keys to delete. The primary keys are case classes. So long as the case class corresponds with the key in Dynamo then you can pass in whatever you want to.

Reply:
Sorry, I realised that I made a mistake in how I formatted the case classes above.

The reason why I was asking was because it wasn't evident from the type signature and documentation, in what format it should be in. I initially assumed it should be like this:

```
List(PrimaryKey("value1"), PrimaryKey("value2"))
```
and then if there are more than 25, it would send them in batches

but noticed from the tests that you can do this too
```
List(PrimaryKeys("value1", "value2")) // (i.e. just put all of the values in one case class)

```

so I thought that one of these methods would either be incorrect or less efficient than the other, especially since we are batching. So if you do this

```
List(PrimaryKeys("key1", "key2" ... "key100"))
```
wouldn't that mean that we are asking dynamo to delete more than 25 delete items at one time? 

Please let me know if I'm not clear, it's hard to explain this; I might have the wrong understanding of how deletions work.

Reply:
Each case class is one primary key. So for the files table it's `FilesTablePrimaryKey(partitionKey: FilesTablePartitionKey, sortKey: FilesTableSortKey)` and for the lock table it's `LockTablePartitionKey(ioId: UUID)` Each case class in the list is one primary key so in the example of `List(PrimaryKeys("key1", "key2" ... "key100"))` that would be one primary key with 100 attributes (not possible I know) to delete one row. There's no way of validating that the case class is a valid primary key though, that's up to the calling code to do.

Reply:
Ah, I see.